client/src/modules/vouchers/modals/search.modal.js

```diff
@@ -13,32 +14,67 @@ VoucherRegistrySearchModalController.$inject = [
  * returning it as a JSON object to the parent controller.  The data can be
  * preset by passing in a filters object using filtersProvider().
  */
-function VoucherRegistrySearchModalController(ModalInstance, filters, Notify, moment, Periods, Store, util) {
+function VoucherRegistrySearchModalController(ModalInstance, filters, Notify, moment, Periods, Store, util, 
+  TransactionTypes, $translate, Vouchers) {
   var vm = this;
   var changes = new Store({ identifier : 'key' });
   var searchQueryOptions = [
     'reference', 'description', 'user_id', 'type_ids',
   ];
 
   vm.filters = filters;
+  // searchQueries is the same id:value pair
   vm.searchQueries = {};
+
   vm.defaultQueries = {};
 
+  var lastViewFilters = Vouchers.filters.formatView().customFilters;
+
+  // map key to last display value for lookup in loggedChange
+  var lastDisplayValues = lastViewFilters.reduce(function (object, filter) {
+    object[filter._key] = filter.displayValue;
+    return object;
+  }, {});  
 
+  // displayValues will be an id:displayValue pair
+  var displayValues = {};
+  
   // assign already defined custom filters to searchQueries object
   vm.searchQueries = util.maskObjectFromKeys(filters, searchQueryOptions);
 
+  // load all Transaction types
+  TransactionTypes.read()
+    .then(function (types) {
+      types.forEach(function (item) {
+        item.typeText = $translate.instant(item.text);
+      });
+      vm.transactionTypes = types;
+    })
+    .catch(Notify.handleError);
+
   if (filters.limit) {
     vm.defaultQueries.limit = filters.limit;
   }
 
   vm.onTransactionTypesChange = function onTransactionTypesChange(transactionTypes) {
     vm.searchQueries.type_ids = transactionTypes;
+    var typeText = '/';
+
+    transactionTypes.forEach(function (typeId) {
```

Comment:
This solution is okay for now.

What we should probably do is modify the `onTransactionTypeChange()` callback to send back the list of transaction types, rather than a list of IDs.  That way we can simply use their labels instead of downloading transaction types twice.